### ISSUE TITLE
[BUG]AssignLeaflets failed when working with a system using CHARMM forcefield

### ISSUE THREAD

**Issue by VOD555** [p-j-smith/lipyphilic#73]
**Describe the bug**
Get an Index Error when running AssignLeaflets with a system built with CHARMM forcefield. The membrane contains POPE and POPG.

**To Reproduce**

``` python

import MDAnalysis as mda
from lipyphilic.lib.assign_leaflets import AssignLeaflets

u = mda.Universe('md.tpr','md.xtc')

leaflets = AssignLeaflets(
    universe=u,
    lipid_sel="name P"
)
leaflets.run(start=None, stop=None, step=None)

```

**Expected behaviour**
Run smoothly and store results in leaflets.leaflets attribute

**Actual behaviour**
Get an Index Error 

```
IndexError                                Traceback (most recent call last)
<ipython-input-7-a7e32bcee85f> in <module>
      3     lipid_sel="name P"
      4 )
----> 5 leaflets.run(start=None, stop=None, step=None)

~/miniconda3/envs/py36/lib/python3.6/site-packages/lipyphilic/lib/base.py in run(self, start, stop, step, verbose)
    192             self.times[i] = ts.time
    193             # logger.info("--> Doing frame {} of {}".format(i+1, self.n_frames))
--> 194             self._single_frame()
    195         logger.info("Finishing up")
    196         self._conclude()

~/miniconda3/envs/py36/lib/python3.6/site-packages/lipyphilic/lib/assign_leaflets.py in _single_frame(self)
    395 
    396         # Assign leaflets
--> 397         self._assign_leaflets(memb_midpoint_xy)
    398 
    399         # if necessary, find midplane residues

~/miniconda3/envs/py36/lib/python3.6/site-packages/lipyphilic/lib/assign_leaflets.py in _assign_leaflets(self, memb_midpoint_xy)
    426         upper_leaflet = self.membrane[
    427             self.membrane.positions[:, 2] >
--> 428             (memb_midpoint_xy.statistic[lipid_x_bins, lipid_y_bins])  # we don't to consider midplane_cutoff here
    429         ]
    430         self.leaflets[

IndexError: index 2 is out of bounds for axis 0 with size 2
```

**Additional context**

- Which version of lipyphilic are you using?
0.6.3
- Which version of Python are you using?
Python 3.6
- Which OS are you using?
Ubuntu 16.04.7 LTS

**Comment by p-j-smith** [comment]
Hi @VOD555 thanks for raising this issue. I'm not immediately sure what the problem is, but I'll have some time this afternoon to look into it. Is there any chance you could share a frame of your trajectory for which the error is raised (I understand if not)?

**Comment by VOD555** [comment]
[step5_input.zip](https://github.com/p-j-smith/lipyphilic/files/6904137/step5_input.zip)
Here's the frame.

**Comment by p-j-smith** [comment]
**tl;dr**

Once I've released v0.8 of lipyphilic (which I'll to this evening), this will work for you:

```
import MDAnalysis as mda
from lipyphilic.lib.assign_leaflets import AssignLeaflets
from lipyphilic.transformations import triclinic_to_orthorhombic

u = mda.Universe('md.tpr','md.xtc')

membrane = u.select_atoms("name P")
u.trajectory.add_transformations(triclinic_to_orthorhombic(ag=membrane))

leaflets = AssignLeaflets(
    universe=u,
    lipid_sel="name P"
)
leaflets.run(start=None, stop=None, step=None)

```


**Explanation**

I noticed you have a triclinic system. However, some analyses in lipyphilic currently do not support triclinic systems.

This is for various reasons and depends on the analysis. For some, such as the `nojump` transformation, it will be possible and probably desirable to explicitly support triclinic systems in the future.

Others, such as `AssignLeaflets`, create a surface of the membrane plane using a two-dimensional rectangular grid. However, I'm not sure it's possible to represent the membrane plane using a rectangular grid for a triclinic system. So, I've added an on-the-fly transformation to convert triclinic coordinates into their orthorhombic representation. This is equivalent to using `gmx trjconv` with the flag `-ur rect`. 

A couple of things to note about the transformation:
    * like `gmx trjconv -ur rect`, all atoms will be put into the primary (orthorhombic) unit cell. This means if your molecules were originally made whole, they will be split across periodic boundaries
    * for this reason, if you are applying multiple on-the-fly transformations, `triclinic_to_orthorhombic` must be the first applied
    * after the coordinates have been modified, the dimensions of your unit cell will also be modified to reflect that you now have an orthorhombic system. Therefore, any atoms you would like to include in your analyses should be included in the `ag` you pass to `triclinic_to_orthorhombic`

I applied the transformation to your system and then used `AssignLeaflets`. Here are the x and z coordinates of the P atoms in your bilayer:

<img src="https://user-images.githubusercontent.com/29753790/127750464-819cd3f7-85ef-4b3c-afe6-a00e43b23fc6.png" width="450">

So you should now be able to use all the tools in lipyphilic without any problems.

Thanks again for raising the issue - triclinic systems are something I had completely neglected, so it's nice to have (indirectly) added support for them.
